src/simplify_qdq.cpp

```diff
@@ -0,0 +1,143 @@
+}
+
+struct match_find_quantizable_ops
+{
+
+    static auto dequantizelinear_op(const std::string& name, const std::string& scale)
+    {
+        return match::name("dequantizelinear")(
+            match::arg(0)(match::skip(match::name("quantizelinear"))(match::any().bind(name))),
+            match::arg(1)(match::skip_broadcasts(has_same_value().bind(scale))),
+            match::arg(2)(match::skip_broadcasts(match::all_of(match::has_value(0)))));
+    }
+
+    auto matcher() const
+    {
+        return match::name(get_quantizable_op_names())(
+            match::arg(0)(dequantizelinear_op("x1", "scale1")),
+            match::arg(1)(dequantizelinear_op("x2", "scale2")));
+    }
+
+    void apply(module& m, match::matcher_result r) const
+    {
+        auto qop    = r.result;
+        auto q1     = r.instructions["x1"];
+        auto q2     = r.instructions["x2"];
+        auto scale1 = r.instructions["scale1"];
+        auto scale2 = r.instructions["scale2"];
+
+        // Only INT8 type currently supported
+        if(q1->get_shape().type() != migraphx::shape::int8_type or
+           q2->get_shape().type() != migraphx::shape::int8_type)
+            return;
+
+        double scale;
+        visit_all(scale1->get_literal(), scale2->get_literal())(
+            [&](const auto s1, const auto s2) { scale = s1.front() * s2.front(); });
+
+        auto qop_args  = qop->inputs();
+        qop_args.at(0) = q1;
+        qop_args.at(1) = q2;
+        instruction_ref dq;
+        instruction_ref dq_scale;
+        instruction_ref zero_point;
+        if(qop->name() == "convolution")
+        {
+            auto conv_val = qop->get_operator().to_value();
+            dq            = m.insert_instruction(
+                qop, migraphx::make_op("quant_convolution", conv_val), qop_args);
+            dq_scale   = m.add_literal(static_cast<float>(scale));
+            zero_point = m.add_literal(0);
+        }
+        else if(qop->name() == "dot")
+        {
+            auto dot_op    = any_cast<op::dot>(qop->get_operator());
+            auto scale_val = dot_op.alpha / scale;
+            auto alpha     = 1;
+            auto beta      = (qop->inputs().size() == 3) ? dot_op.beta : 0.0f;
+            dq             = m.insert_instruction(
+                qop, migraphx::make_op("quant_dot", {{"alpha", alpha}, {"beta", beta}}), qop_args);
+            dq_scale   = m.add_literal(static_cast<float>(scale_val));
+            zero_point = m.add_literal(static_cast<int>((-1.0f * beta) / scale_val));
+        }
+
+        auto lens = dq->get_shape().lens();
+        auto scale_mb =
+            m.insert_instruction(qop, make_op("multibroadcast", {{"output_lens", lens}}), dq_scale);
+        auto shift_mb = m.insert_instruction(
+            qop, make_op("multibroadcast", {{"output_lens", lens}}), zero_point);
+        dq = m.insert_instruction(qop, make_op("dequantizelinear"), dq, scale_mb, shift_mb);
+        m.replace_instruction(qop, dq);
```

Comment:
I would suggest create a unit test as (pseudo code):

```

a = m.add_parameter("a", sa);
b = m.add_parameter("b", sb);
c = m.add_parameter("c", sc);
sa = 0.33f;
sb = 0.72f;
qa = m.add_instruction(make_op("quantizelinear"), a, sa);
dqa = m.add_instruction(make_op("quantizelinear"), qa, sa); 
qb = m.add_instruction(make_op("quantizelinear"), b, sb);
dqb = m.add_instruction(make_op("quantizelinear"), qb, sb); 
auto d = m.add_instruction(make_op("dot", {{"alpha", 0.3}, {"beta", 0.5}}, dqa, dqb, c)
m.add_return({d});
```

For the above program:
case 1) replace dot with quant_dot, create an input, run the program, get result1
case 2), remove the qdq for both a and b, create the same input as in case 1), run the program, get result 2

result1 should be equal to result2

Reply:
> I would suggest create a unit test as (pseudo code):

I would prefer not to add a test case like that with `dot` having non-default alpha and beta since we are in the process of removing these.

Reply:
Should I still add a similar test case, but with default alpha and beta?